projects/openapi-utilities/src/openapi/openapi2/openapi-2-traverser.ts

```diff
@@ -0,0 +1,891 @@
+          },
+          value,
+        };
+    }
+  }
+
+  *traverseResponseHeaders(
+    response: OpenAPIV2.ResponseObject,
+    jsonPath: string,
+    conceptualPath: IPathComponent[],
+    location: ResponseLocation
+  ): IterableIterator<IFact> {
+    if (response.headers) {
+      for (let [name, header] of Object.entries(response.headers)) {
+        const nextJsonPath = jsonPointer.append(jsonPath, 'headers', name);
+        if (!isObject(header)) {
+          console.warn(
+            `Expected an object at: ${getReadableLocation(
+              nextJsonPath
+            )}, found ${header}`
+          );
+        } else if (isNotReferenceObject(header)) {
+          yield this.onResponseHeader(
+            name,
+            header,
+            nextJsonPath,
+            [...conceptualPath, 'headers', name],
+            {
+              ...location,
+              inResponse: {
+                statusCode: location.inResponse.statusCode,
+                header: name,
+              },
+            }
+          );
+        } else {
+          console.warn(
+            `Expected a flattened spec, found a reference at: ${getReadableLocation(
+              nextJsonPath
+            )}`
+          );
+        }
+      }
+    }
+  }
+
+  onResponseHeader(
+    name: string,
+    header: OpenAPIV2.HeaderObject,
+    jsonPath: string,
+    conceptualPath: IPathComponent[],
+    location: ResponseHeaderLocation
+  ): FactVariant<OpenApiKind.ResponseHeader> {
+    this.checkJsonTrail(jsonPath, header);
+    const value = {
+      name,
+      ...header,
+    };
+    return {
+      location: {
+        jsonPath,
+        conceptualPath,
+        kind: OpenApiKind.ResponseHeader,
+        conceptualLocation: location,
+      },
+      value,
+    };
+  }
+
+  *traverseBody(
+    body: { schema?: OpenAPIV2.SchemaObject },
+    contentType: string,
+    jsonPath: string,
+    conceptualPath: IPathComponent[],
+    location: BodyLocation
+  ): IterableIterator<IFact> {
+    this.checkJsonTrail(jsonPath, body);
+
+    const { schema } = body;
+
+    if (schema) {
+      const nextJsonPath = jsonPointer.append(jsonPath, 'schema');
+      if (!isObject(schema)) {
+        console.warn(
+          `Expected an object at: ${getReadableLocation(
+            nextJsonPath
+          )}, found ${schema}`
+        );
+      } else if (isNotReferenceObject(schema)) {
+        yield this.onContentForBody(
+          schema,
+          contentType,
+          jsonPath,
+          conceptualPath,
+          location
+        );
+        yield* this.traverseSchema(schema, nextJsonPath, conceptualPath, {
+          ...location,
+          jsonSchemaTrail: [],
+        });
+      } else {
+        console.warn(
+          `Expected a flattened spec, found a reference at: ${getReadableLocation(
+            nextJsonPath
+          )}`
+        );
+      }
+    }
+
+    // @ limitation no example support
```

Comment:
> Is this a limitation because we haven't coded it here 
yes 

> ...or we don't handle examples in a way that works with OAS2
not that, but our current oas3 piece is incomplete too. I wouldn't feel safe saying optic supports examples in either spec yet. It's not 100% clear to me that we need seperate examples or if they're part of the entity they are attached to. 

The sorts of rules teams want for examples are things like
- example matches schema 
- has example
- example includes optional properties (not just required) 

We've never really supported examples fully in oas3. We allowed them in response and request bodies, but not on individual properties and a handful of other places the spec allows for. They work differently in oas2 -- we should probably make some decisions about how much we want to support in both versions. 

https://blog.stoplight.io/difference-between-open-v2-v3-v31